File: frontend/rename_assets.js
// frontend/rename_assets.js
const fs = require("fs");
const path = require("path");

const outDir = path.join(__dirname, "out");
const nextDir = path.join(outDir, "_next");
const targetDir = path.join(outDir, "next");

// 1. Rename _next to next
if (fs.existsSync(nextDir)) {
  if (fs.existsSync(targetDir)) {
    fs.rmSync(targetDir, { recursive: true, force: true });
  }
  fs.renameSync(nextDir, targetDir);
  console.log("Renamed _next to next in out folder!");
} else {
  console.log("_next folder not found or already renamed.");
}

// 2. Recursively replace paths in files
function replaceInFiles(dir) {
  const list = fs.readdirSync(dir);
  list.forEach(file => {
    const filePath = path.join(dir, file);
    const stat = fs.statSync(filePath);
    
    if (stat.isDirectory()) {
      replaceInFiles(filePath);
    } else if (stat.isFile() && /\.(html|js|css|json|txt)$/.test(file)) {
      try {
        let content = fs.readFileSync(filePath, "utf8");
        const updated = content.replace(/\/_next\//g, "/next/").replace(/_next\//g, "next/");
        
        if (updated !== content) {
          fs.writeFileSync(filePath, updated, "utf8");
          console.log(`Updated paths in: ${file}`);
        }
      } catch (e) {
        console.error(`Error processing ${file}: ${e}`);
      }
    }
  });
}

replaceInFiles(outDir);

File: frontend/rename_assets.py
# frontend/rename_assets.py
import os
import shutil

out_dir = os.path.join(os.path.dirname(__file__), "out")
next_dir = os.path.join(out_dir, "_next")
target_dir = os.path.join(out_dir, "next")

if os.path.exists(next_dir):
    if os.path.exists(target_dir):
        shutil.rmtree(target_dir)
    shutil.move(next_dir, target_dir)
    print("Renamed _next to next in out folder!")
else:
    print("_next folder not found or already renamed.")

# Search and replace all occurrences of _next with next in output files
for root, dirs, files in os.walk(out_dir):
    for file in files:
        if file.endswith((".html", ".js", ".css", ".json", ".txt")):
            file_path = os.path.join(root, file)
            try:
                with open(file_path, "r", encoding="utf-8") as f:
                    content = f.read()
                
                # Replace paths
                new_content = content.replace("/_next/", "/next/").replace("_next/", "next/")
                
                if new_content != content:
                    with open(file_path, "w", encoding="utf-8") as f:
                        f.write(new_content)
                    print(f"Updated paths in: {file}")
            except Exception as e:
                print(f"Error processing {file}: {e}")
